pkg/await/extensions_ingress.go

```diff
@@ -327,7 +327,9 @@ func (iia *ingressInitAwaiter) errorMessages() []string {
 
 	if !iia.ingressReady {
 		messages = append(messages,
-			"Ingress was not allocated an IP address; does your cloud provider support this?")
+			"Ingress .status.loadBalancer field was not updated with a hostname/IP address; "+
+				"If you're using the Traefik ingress-controller, make sure it is configured with the "+
```

Comment:
While helpful, I'm not sure we should be guessing about specific causes like this unless we think those are 90% cases.  Can we generalize this at all to make it suggestive of the same root cause and fix, but while applying a larger class of causes?  Or should we start maintaining a Kubernetes provider troubleshooting docs page where we can include this kind of helpful text in longer form and link to it from the error messages here?

Reply:
Curious @hausdorff thoughts on this as well.

Reply:
For example - could we add something like "Ensure that your ingress controller is configured to update the `status` section of ingress objects"?  (Inspired by the text at https://docs.traefik.io/configuration/backends/kubernetes/)

Reply:
Yeah, that's reasonable. One of the previous platforms I worked on would put a link in the error message to a knowledge base page we maintained. That way you could still get detailed troubleshooting info like that, but it was easier to maintain.

Reply:
Yeah - we have https://pulumi.io/reference/troubleshooting.html which is used for this purpose by the CLI for core CLI errors.  We could extend that to have sections for other providers to add error message help as well, or could create dedicated troubleshooting pages under the `/kubernetes` section.

Reply:
It sounds like we all agree about what to do here -- but to make it explicit, my preference is also to make error messages like this one follow a pattern:

* Factual explanation of what is wrong (_i.e._, ingress controller not configured to emit status)
* Link to a docs/playbooks/whatever

There are many reasons why I think this is a useful pattern -- lets us evolve recommended mitigations gradually and completely independently as we mature; not tractable to explain all mitigations in the error message; docs site experience is going to be way better for most complex issues; _etc_.